test/test_decomp.py

```diff
@@ -406,6 +407,111 @@ def test_uniform(self, device):
         res = torch._decomp.decompositions.uniform(x, low=low, high=high)
         self.assertEqual(ref, res)
 
+
+    @unittest.skipIf(TEST_WITH_ASAN, "Skipped under ASAN")
+    @suppress_warnings
+    @modules(module_db)
+    def test_module(self, device, dtype, module_info, training):
+        module_cls = module_info.module_cls
+        is_python_dispatcher_decomp = module_cls == torch.nn.RNN
+        module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
+                                                       requires_grad=True, training=training)
+        for module_input in module_inputs:
+            if module_input.forward_input is None:
+                continue
+            args, kwargs = module_input.constructor_input.args, module_input.constructor_input.kwargs
+            m = module_cls(*args, **kwargs)
+            m.to(device).to(dtype)
+
+            args, kwargs = module_input.forward_input.args, module_input.forward_input.kwargs
+            with self.DecompCrossRefMode(self, self.precision, self.rel_tol, dtype), enable_python_dispatcher():
+                decomp_out = m(*args, **kwargs)
+
+            non_decomp_out = m(*args, **kwargs)
+            if is_python_dispatcher_decomp:
+                # without this check, incorrect decomps at the python dispatcher level can still pass because
+                # they're checking aten decomps at the
```

Comment:
Comment cut off -- why is is_python-dispatcher_decoomp necessary?

Reply:
Ahhh yeah so this is what I pieced together that the testing did:
(1) runs the python dispatcher
(2) runs a mode that checks that the decompositions at the python key level match their non-decomposed versions

It doesn't check that the overall function matches the original function without the python dispatcher or any decompositions matches the decomposed version. I think this because if there's small numerical differences, they might get magnified. However, since we're mostly testing the python dispatcher decomp here (not the decomposition of linear or something), I do specifically want to check that these decompositions match. To your other points, we wouldn't need this if this was a post-autograd decomposition only

Originally I had figured that we could use the python dispatcher directly in expanded weights but since we aren't doing that I guess we could have these be post autograd decompositions instead and we wouldn't need this

Reply:
It sounds like we're worried about small numeric differences and tolerances. If that is the case, can we always test in float64 and remove the special case for `is_python_dispatcher_decomp`? The special case makes it so that the test isn't very ModuleInfo-agnostic; someone updating it in the future needs to update the condition

Reply:
Ah sorry I wasn't clear. So the issue is that small differences get magnified when they're propagated. This happens for float64 and 32 and the differences end up being rather significant (abs diff of sometimes like 1.5, relative difference of like 8). The rest of the testing has this issue but it's picked up by proxy tensor tests (they're just harder to write for modules). The errors I was seeing were all for the RNNCells. Since there aren't any decompositions for the RNNCells, it seemed more like a testing bug 

But I totally hear you on the special case, a different option that might be nicer is instead of testing all the modules only test RNNs since we've only written decomps for RNNs). It's a special case but a special case in the test filter. I just put up this change for this but can revert if the previous version is preferred